db/entity/src/conversions/channels.rs

```diff
@@ -45,23 +48,32 @@ impl TryFrom<&channel::Model> for ChannelEntry {
     type Error = DbEntityError;
 
     fn try_from(value: &channel::Model) -> Result<Self, Self::Error> {
-        let status = value.try_into()?;
-        Ok(ChannelEntry::new(
-            value.source.parse()?,
-            value.destination.parse()?,
-            HoprBalance::from(U256::from_be_bytes(&value.balance)),
-            U256::from_be_bytes(&value.ticket_index),
-            status,
-            U256::from_be_bytes(&value.epoch),
-        ))
+        if value.corrupted {
+            return Err(DbEntityError::ConversionError(
+                "cannot convert corrupted channel model to ChannelEntry".into(),
+            ));
+        }
+        match model_to_channel_entry_unchecked(&value) {
+            Ok(channel_entry) => Ok(channel_entry),
+            Err(e) => Err(e),
+        }
     }
 }
 
 impl TryFrom<channel::Model> for ChannelEntry {
     type Error = DbEntityError;
 
     fn try_from(value: channel::Model) -> Result<Self, Self::Error> {
-        (&value).try_into()
+        if value.corrupted {
+            return Err(DbEntityError::InvalidCorruptionFlag(
+                "cannot convert corrupted channel model to ChannelEntry".into(),
+            ));
+        }
+
+        match model_to_channel_entry_unchecked(&value) {
+            Ok(channel_entry) => Ok(channel_entry),
+            Err(e) => Err(e),
+        }
```

Comment:
I think there was a misunderstanding, I didn't mean to make this conversion explicit

```suggestion
        (&value).try_into()
```